src/V3Begin.cpp

```diff
@@ -144,9 +144,19 @@ class BeginVisitor final : public VNVisitor {
 
     // VISITORS
     void visit(AstFork* nodep) override {
-        // Keep this begin to group its statements together
+        // Keep begins in forks to group their statements together
         VL_RESTORER(m_keepBegins);
         m_keepBegins = true;
+        // If a statement is not a begin, wrap it in a begin. This fixes an issue when the
+        // statement is a task call that gets inlined later (or any other statement that gets
+        // replaced with multiple statements)
+        for (AstNode* stmtp = nodep->stmtsp(); stmtp; stmtp = stmtp->nextp()) {
```

Comment:
The overall seems conceptually reasonable.  

Note here you are iterating over a list that is changing, you need to make a nextp=stmtp->nextp() before editing, then use nextp in the loop.

Reply:
Right, I was planning to update `stmtp` actually, but forgot to do it. Or do you prefer adding `nextp`?